Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -7,401 +7,882 @@
+                        }
+                    }
+                }
+            }
+
+            ++allocationCount;
+
+            return true;
+        };
+
+        // GetAllocationCount runs EnumerateAllocations for the only the first allocation record
+        // in order to get the current allocation count at the time of the call
+        size_t estimateAllocationCount{};
+        auto GetAllocationCount = [&estimateAllocationCount](void*, const Debug::AllocationInfo&, unsigned int, size_t numRecords)
+        {
+            estimateAllocationCount = numRecords;
+            return false;
+        };
+
+        // Stores an estimate of how many allocation records can be output per minute
+        // This is based on empirical data of dumping the SystemAllocator in the Editor
+        // About 10000 records can be printed per second
+        constexpr size_t AllocationRecordsPerSecondEstimate = 10000;
+        using SecondsAsDouble = AZStd::chrono::duration<double>;
+
+        // Iterate over each allocator to dump and print their allocation records
+        for (AZ::IAllocator* allocator : allocatorsToDump)
+        {
+            AZ::Debug::AllocationRecords* records = allocator->GetRecords();
+
+            estimateAllocationCount = 0;
+            // Get the allocation count at the time of the first EnumerateAllocations call
+            // NOTE: This is only an estimatation of the count as the number of allocation
+            // can change between this call and the next call EnumerationAllocations to print the records
+            records->EnumerateAllocations(GetAllocationCount);
+
+            auto allocationRecordsSecondsEstimate = SecondsAsDouble(estimateAllocationCount / double(AllocationRecordsPerSecondEstimate));
+
+            AllocationString printString = AllocationString::format(
+                R"(Printing allocation records for allocator %s. Estimated time to print all records is %lf seconds)"
+                "\n",
+                allocator->GetName(),
+                allocationRecordsSecondsEstimate.count());
+            printStream.Write(printString.size(), printString.c_str());
+            // Also write the print string above to the Trace system
+            AZ::Debug::Trace::Instance().Output(MemoryTag, printString.c_str());
+
+            allocationCount = 0;
+            auto startTime = AZStd::chrono::steady_clock::now();
+            records->EnumerateAllocations(PrintAllocations);
+            auto endTime = AZStd::chrono::steady_clock::now();
+            auto durationInMilliseconds = AZStd::chrono::duration_cast<AZStd::chrono::milliseconds>(endTime - startTime);
+            auto durationInSecondsAsDouble = AZStd::chrono::duration_cast<SecondsAsDouble>(durationInMilliseconds);
+
+            AllocationString allocationsPerSecond = u8"\u221e";
+            if (durationInSecondsAsDouble.count() != 0)
+            {
+                // Use a double for the seconds duration in order to allow decimal values
```

Comment:
I think it would be ok to just round up to the nearest second, especially since it's just a rough estimate, but decimals works too. :)

Reply:
Yeah for allocators with a small number of records, this was outputting 0 seconds, but we can round up to the nearest second in that case